kernel/sem.c

```diff
@@ -69,9 +69,9 @@ int z_impl_k_sem_init(struct k_sem *sem, unsigned int initial_count,
 		return -EINVAL;
 	}
 
-	sys_trace_void(SYS_TRACE_ID_SEMA_INIT);
 	sem->count = initial_count;
 	sem->limit = limit;
+	sys_trace_semaphore_init(sem);
```

Comment:
I understand the change to `sys_trace_semaphore_init(sem)`. Why did we move it below the `sem-`>` assignments? In this position, does it better represent the timing of the trace at initializing the semaphore (as in the next line z_waitq_init)?

Reply:
> Why did we move it below the `sem-`>` assignments?

to get the values the semaphore was initiated with, this could be part of the tracing later.